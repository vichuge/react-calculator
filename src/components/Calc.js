import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

const Calc = ({ total: initialTotal, next: initialNext, operation: initialOperation }) => {
  const [total, setTotal] = useState(initialTotal);
  const [next, setNext] = useState(initialNext);
  const [operation, setOperation] = useState(initialOperation);

  const handleClick = (buttonName) => {
    const result = calculate({ total, next, operation }, buttonName);
    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  };

  let display = '';
  if (!total) display = '0';
  if (total) display = total;
  if (total && operation) display = total;
  if (total && operation && next) display = next;

  return (
    <div className="calc-page">
      <div className="Text">
        <h2>Lets do some Math!</h2>
      </div>
      <div className="calculator">
        <Display result={display} />
        <ButtonPanel clickApp={handleClick} />
      </div>
    </div>
  );
};

Calc.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
  operation: PropTypes.string,
};

Calc.defaultProps = {
  total: null,
  next: null,
  operation: null,
};

export default Calc;
