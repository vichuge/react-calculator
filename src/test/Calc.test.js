import React from 'react';
import renderer from 'react-test-renderer';
import Calc from '../components/Calc';

it('return display', () => {
  const tree = renderer
    .create(<Calc />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
