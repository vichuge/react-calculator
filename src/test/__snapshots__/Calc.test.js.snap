// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`return display 1`] = `
<div
  className="calc-page"
>
  <div
    className="Text"
  >
    <h2>
      Lets do some Math!
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <p>
        0
      </p>
    </div>
    <div
      className="g1"
    >
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange "
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="g1"
    >
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange "
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="g1"
    >
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange "
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="g1"
    >
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange "
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="g1"
    >
      <button
        className="zero "
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="gray "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange "
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
